Datahub.Web/Elasticsearch/ElasticsearchService.cs

```diff
@@ -0,0 +1,118 @@
+
+        /// <summary>
+        /// Initialises a new ElasticClient instance with support for localhost and AWS endpoints.
+        /// </summary>
+        public ElasticsearchService()
+        {
+            var builder = new UriBuilder();
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("ELASTICSEARCH_HOST_SCHEME"))) 
+            {
+                builder.Scheme = Environment.GetEnvironmentVariable("ELASTICSEARCH_HOST_SCHEME");
+            }
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("ELASTICSEARCH_HOST_PORT")))
+            {
+                builder.Port = int.Parse(Environment.GetEnvironmentVariable("ELASTICSEARCH_HOST_PORT"));
+            }
+            builder.Host = Environment.GetEnvironmentVariable("ELASTICSEARCH_HOST");
+
+            var endpointUri = new Uri(builder.ToString());
+            var pool = new SingleNodeConnectionPool(endpointUri);
+
+            string awsAccessKey = Environment.GetEnvironmentVariable("ELASTICSEARCH_AWS_ACCESSKEY");
+            string awsSecretAccessKey = Environment.GetEnvironmentVariable("ELASTICSEARCH_AWS_SECRETACCESSKEY");
+            string awsRegion = Environment.GetEnvironmentVariable("ELASTICSEARCH_AWS_REGION");
+            string awsProfile = Environment.GetEnvironmentVariable("ELASTICSEARCH_AWS_PROFILE");            
+
+            if (endpointUri.IsLoopback)
+            {
+                // support localhost for local development
+                _client = new ElasticClient(new ConnectionSettings(pool));
+            }
+            else if (!string.IsNullOrWhiteSpace(awsAccessKey) && !string.IsNullOrWhiteSpace(awsSecretAccessKey))
+            {
+                // use AWS access keys to configure
+                var httpConnection = new AwsHttpConnection(awsRegion, new StaticCredentialsProvider(new AwsCredentials
+                {
+                    AccessKey = awsAccessKey,
+                    SecretKey = awsSecretAccessKey
+                }));
+                _client = new ElasticClient(new ConnectionSettings(pool, httpConnection));
+            }
+            else if (!string.IsNullOrWhiteSpace(awsProfile))
+            {
+                // use AWS profile
+                throw new NotImplementedException("AWS profile support is not working yet. Specify access keys instead.");
+                // var httpConnection = new AwsHttpConnection(awsRegion, new NamedProfileCredentialProvider(awsProfile));
+                // _client = new ElasticClient(new ConnectionSettings(pool, httpConnection));
+            }
+            else
+            {
+                // attempt to use AWS instance profile (for production)
+                var httpConnection = new AwsHttpConnection(awsRegion, new InstanceProfileCredentialProvider());
+                _client = new ElasticClient(new ConnectionSettings(pool, httpConnection));
+            }
+        }
+
+        public ElasticClient Client()
```

Comment:
This should be a property rather than a method - in .NET methods are generally verbs. If something just returns the same object reference whenever it's called then it's a property.